From 4b589a8cbc520a588e3968fed20a47d620cbc1cb Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Thu, 11 Feb 2016 11:26:22 +0100
Subject: [PATCH 15/17] tty: imx: Fix dma buffer overflow for low latency

The driver uses the dma if the option UPF_LOW_LATENCY is given to the
interface by the userspace tool. This should not be done, because then
the driver detects buffer overflows in the dma buffer.

Ticket #937
---
 drivers/tty/serial/imx.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index d557f1e..a0493b3 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -1213,7 +1213,8 @@ static int imx_startup(struct uart_port *port)
 
 	/* Can we enable the DMA support? */
 	if (is_imx6q_uart(sport) && !uart_console(port)
-		&& !sport->dma_is_inited)
+		&& !sport->dma_is_inited
+		&& !sport->port.flags == UPF_LOW_LATENCY)
 		imx_uart_dma_init(sport);
 
 	if (sport->dma_is_inited) {
-- 
1.9.1

